src/lib.rs

```diff
@@ -288,18 +289,11 @@ pub fn normalized_levenshtein(a: &str, b: &str) -> f64 {
 /// assert_eq!(3, osa_distance("ab", "bca"));
 /// ```
 pub fn osa_distance(a: &str, b: &str) -> usize {
-    let a_len = a.chars().count();
     let b_len = b.chars().count();
-    if a == b {
```

Comment:
This special case would help performance if the two strings are similar, but hurt if they are not.

Generally it would be better to reduce the common prefix + postfix and return early in case the remaining length is 0. This has the same runtime, but always either helps the performance or doesn't really hurt. In our case the added binary size isn't really worth it though.

Reply:
I would also guess that it's rare to actually hit these cases in real world usage.